File: hashing/subset_arr.cpp
#include <iostream>
#include <vector>
#include <unordered_set>

using namespace std;

/* Check whether B is a subset of A*/
bool isSubset(vector<int> A, vector<int> B) {

    unordered_set<int> keys;

    keys.insert(A.begin(), A.end());

    for (int i = 0; i < B.size(); i++) {
        if (keys.find(B[i]) == keys.end()) {
            return false;
        }
    }

    return true;
}

int main(int argc, char const *argv[])
{   
    vector<int> arr1 = {10, 5, 2, 23, 19};
    vector<int> arr2 = {19, 5, 3};

    cout << isSubset(arr1, arr2) << endl;

    return 0;
}

File: hashing/subset_arr.java
import java.util.HashSet;

public class Hashing {
    
    /*
        Function to return whether array B is a subset of array A
    */
    public static boolean isSubset(int[] A, int[] B) {

        HashSet<Integer> keys = new HashSet<Integer>();

        for (int i = 0; i < A.length; i++) {

            if (!keys.contains(A[i])) {
                keys.add(A[i]);
            }

        }

        for (int i = 0; i < B.length; i++) {

            if (!keys.contains(B[i])) {
                return false;
            }

        }

        return true;

    }

    /*
        Driver function to test the hashing function 
    */
    public static void main(String[] args) {

        int arr1[] = {10, 5, 2, 23, 19};
        int arr2[] = {19, 5, 3};

        System.out.print(isSubset(arr1, arr2) + "\n");

    }
}